Tasks/KubernetesV1/src/clusterconnection.ts

```diff
@@ -64,14 +66,24 @@ export default class ClusterConnection {
             kubeconfig = await this.getKubeConfig(connectionType);
         }
 
-        return this.initialize().then(() => {
+        return this.initialize().then(async () => {
             if (kubeconfig)
             {
                 this.kubeconfigFile = path.join(this.userDir, "config");
                 fs.writeFileSync(this.kubeconfigFile, kubeconfig);
             }
 
             process.env["KUBECONFIG"] = this.kubeconfigFile;
+
+            try {
+              const kubelogin = new Kubelogin(this.userDir);
+              if (kubelogin.isAvailable()) {
+                tl.debug('Kubelogin is installed. Converting kubeconfig.');
+                await kubelogin.login(tl.getInput('azureSubscriptionEndpoint', false));
+              }
+            } catch (err) {
+              tl.warning(tl.loc('KubeloginFailed', err));
```

Comment:
>warning [](http://example.com/codeflow?start=17&length=7)

I'm thinking this should be an error, otherwise, the deployment action could fail but task status would be success

Reply:
I actually changed it to debug message. If there is an issue with kubelogin configuration the error wll be thrown when kubectl invokes get_token